Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/webapp/WebAppPropertyViewPresenter.java

```diff
@@ -22,25 +22,35 @@
 
 package com.microsoft.tooling.msservices.serviceexplorer.azure.webapp;
 
+import java.util.List;
 import java.util.Map;
 import java.util.Set;
 
+import com.microsoft.azure.management.appservice.DeploymentSlot;
 import com.microsoft.azure.management.appservice.WebAppBase;
 import com.microsoft.azuretools.azurecommons.helpers.NotNull;
+import com.microsoft.azuretools.azurecommons.helpers.Nullable;
 import com.microsoft.azuretools.core.mvp.model.webapp.AzureWebAppMvpModel;
 import com.microsoft.tooling.msservices.serviceexplorer.azure.webapp.base.WebAppBasePropertyViewPresenter;
 
 public class WebAppPropertyViewPresenter extends WebAppBasePropertyViewPresenter {
     @Override
-    protected void performUpdateSettings(@NotNull final String sid, @NotNull final WebAppBase webAppBase,
-                                         final Map toUpdate, final Set toRemove) throws Exception {
-        AzureWebAppMvpModel.getInstance().updateWebAppSettings(sid, webAppBase.id(), toUpdate, toRemove);
+    protected void updateAppSettings(@NotNull final String sid, @NotNull final String webAppId,
+                                     @Nullable final String name, final Map toUpdate,
+                                     final Set toRemove) throws Exception {
+        AzureWebAppMvpModel.getInstance().updateWebAppSettings(sid, webAppId, toUpdate, toRemove);
     }
 
     @Override
-    protected boolean performGetPublishingProfile(@NotNull final String sid, @NotNull final WebAppBase webAppBase,
-                                                  @NotNull final String filePath) throws Exception {
-        return AzureWebAppMvpModel.getInstance().getPublishingProfileXmlWithSecrets(
-            sid, webAppBase.id(), filePath);
+    protected boolean getPublishingProfile(@NotNull final String sid, @NotNull final String webAppId,
+                                           @Nullable final String name,
+                                           @NotNull final String filePath) throws Exception {
+        return AzureWebAppMvpModel.getInstance().getPublishingProfileXmlWithSecrets(sid, webAppId, filePath);
+    }
+
+    @Override
+    protected WebAppBase getWebApp(@NotNull final String sid, @NotNull final String webAppId,
```

Comment:
Why do we need to get a WebApp as WebAppBase

Reply:
Better naming to `getWebAppBase`